### PR TITLE
Add schema encode/decode functionality

### PR DESCRIPTION
**Pull request by soerenbf**
## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: src/schema.rs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    schema_json, to_bytes, Cursor, ParseError,
+};
+use derive_more::From;
+use serde_json::Value;
+
+/// Error type returned by the bridge functions.
+/// A corresponding Swift type will be generated (via the UDL definition).
+#[derive(Debug, thiserror::Error)]
+pub enum SchemaError {
+    /// Failed to convert with schema
```
Comment (DOBEN) [inline src/schema.rs]:
```suggestion
    /// Failed to convert the value with the schema
```

File: src/schema.rs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+#[derive(From)]
+pub struct TypeSchema {
+    pub value: Vec<u8>,
+}
+
+pub fn display_type_schema_template(schema: TypeSchema) -> Result<String, SchemaError> {
+    let value_type: Type = from_bytes(&schema.value)?;
+    let template = value_type.to_json_template();
+    Ok(serde_json::to_string(&template)?)
```
Comment (DOBEN) [inline src/schema.rs]:
```suggestion
    Ok(serde_json::to_string_pretty(&template)?)
```

File: src/schema.rs [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    Ok(serde_json::to_string(&json)?)
+}
+
+pub fn serialize_type_value(json: String, schema: TypeSchema) -> Result<Vec<u8>, SchemaError> {
+    let value_type: Type = from_bytes(&schema.value)?;
+    let json: Value = serde_json::from_str(&json)?;
+    Ok(value_type.serial_value(&json)?)
+}
+
+pub fn get_receive_parameter_schema(
```
Comment (DOBEN) [inline src/schema.rs:96]: It is probably not used by the wallets, but to make the SDK complete in extracting the schemas, a function for `get_event_schema` can be added.

**Review by DOBEN — APPROVED** [review]

File: src/schema.rs (hunk 3 above)
Comment (soerenbf) [reply src/schema.rs:96]: Huh, I didn't know this existed! Thanks 😄

**Pull request merged**